src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,143 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            growIfArrayFull();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexToAdd(index);
+        if (size == elementData.length) {
+            growIfArrayFull();
+        }
+        if (elementData[index] != null) {
+            divideAndSetNewValue(value, index);
+        } else if (elementData[index] == null) {
+            elementData[index] = value;
+        }
```

Comment:
The check for `elementData[index] != null` is unnecessary and incorrect. When adding an element at a specific index, you should always shift the elements to the right, regardless of whether the current element at that index is null or not. This logic should be revised.